lib/rubocop/cop/style/disable_cops_within_source_code_directive.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+# rubocop:disable Lint/UnneededCopDisableDirective
+
+module RuboCop
+  module Cop
+    module Style
+      # Detects comments to enable/disable rubocop.
+      # This is useful if want to make sure that every Rubocop error gets fixed
+      # and not quickly disabled with a comment.
+      #
+      # @example
+      #   # bad
+      #   # rubocop:disable Metrics/AbcSize
+      #   def f
+      #   end
+      #   # rubocop:enable Metrics/AbcSize
+      #
+      #   # good
+      #   def fixed_method_name_and_no_rubocop_comments
+      #   end
+      #
+      class DisableCopsWithinSourceCodeDirective < Cop
+        # rubocop:enable Lint/UnneededCopDisableDirective
+        MSG = 'Comment to disable/enable Rubocop.'
+
+        def investigate(processed_source)
+          processed_source.comments.each do |comment|
+            next unless comment.text.start_with?('# rubocop:')
```

Comment:
There's a constant, `CommentConfig::COMMENT_DIRECTIVE_REGEXP` that you could use instead to catch all variations on whitespace that we allow in these comments.

Reply:
Done